tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -807,3 +808,113 @@ def create_or_delete_until_producer_fin() -> bool:
 
         producer.wait()
         rand_consumer.wait()
+
+
+class EndToEndSpilloverTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3,
+                        cleanup_policy=TopicSpec.CLEANUP_DELETE), )
+
+    def __init__(self, test_context):
+        self.si_settings = SISettings(
+            test_context,
+            log_segment_size=1024,
+            fast_uploads=True,
+            cloud_storage_housekeeping_interval_ms=10000)
+        super(EndToEndSpilloverTest, self).__init__(
+            test_context=test_context,
+            # Set to minimal value
+            extra_rp_conf=dict(
+                cloud_storage_spillover_manifest_max_segments=10),
+            si_settings=self.si_settings)
+
+        self.msg_size = 1024 * 256
+        self.msg_count = 3000
+
+    def produce(self):
+        topic_name = self.topics[0].name
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       topic_name,
+                                       msg_size=self.msg_size,
+                                       msg_count=self.msg_count)
+
+        producer.start()
+        producer.wait()
+        producer.free()
+
+        def all_partitions_spilled():
+            return self.num_manifests_uploaded() > 0
```

Comment:
nit: this check doesn't ensure all partitions have spilled, but we can come back to it